src/components/Menu/MenuItem.tsx

```diff
@@ -147,27 +157,48 @@ class MenuItem extends UIComponent<Extendable<IMenuItemProps>, IMenuItemState> {
       >
         {childrenExist(children) ? (
           children
+        ) : !menu ? (
+          this.renderMenuItem(accessibility, classes)
         ) : (
-          <a
-            className={cx('ui-menu__item__anchor', classes.anchor)}
-            onClick={this.handleClick}
-            onBlur={this.handleBlur}
-            onFocus={this.handleFocus}
-            {...accessibility.attributes.anchor}
-            {...accessibility.keyHandlers.anchor}
+          <Popup
+            open={submenuOpen}
+            align={vertical ? 'top' : 'start'}
+            position={vertical ? 'after' : 'below'}
+            content={{
+              content: <Menu items={menu.items} vertical type={type} />,
+              styles: {
+                padding: '0px',
+                border: '',
+              },
+            }}
+            // content={<Menu items={menu.items} type={type} vertical/>}
```

Comment:
which trouble it creates, specifically?

Please, note that when you are using 
```js
content={<Menu items={menu.items} type={type} vertical/>}
```
you don't introduce any wrapper for the Popup's content - and it could happen that the content that you are providing unwrapped (by the expression mentioned above) could have `z-index` value that would prevent it to be shown **on top** of the page's content. Consider to review this example for the Popup for more details: https://stardust-ui.github.io/react/components/popup#types-popup-content-wrapper-example

Reply:
Ah, totally missed the ```z-index```. Thanks for pointing it out.